consensus/misc/eip1559.go

```diff
@@ -0,0 +1,66 @@
+package misc
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+func VerifyEip1559Header(parent, header *types.Header, notFirst bool) error {
+	// Verify that the gasUsed is <= gasTarget*elasticityMultiplier
+	if header.GasUsed > header.GasLimit*params.ElasticityMultiplier {
+		return fmt.Errorf("exceeded elasticity multiplier: gasUsed %d, gasTarget*elasticityMultiplier %d", header.GasUsed, header.GasLimit*params.ElasticityMultiplier)
+	}
+
+	// Verify the baseFee is correct based on the parent header.
+	expectedBaseFee := new(big.Int).SetUint64(params.InitialBaseFee)
+	if notFirst {
+		// Only calculate the correct baseFee if the parent header is
+		// also a EIP-1559 header.
+		expectedBaseFee = CalcBaseFee(parent)
+	}
+	if header.BaseFee.Cmp(expectedBaseFee) != 0 {
+		return fmt.Errorf("invalid baseFee: expected: %d, have %d", parent.BaseFee.Int64(), header.BaseFee.Int64())
+	}
+
+	return nil
+}
+
+func CalcBaseFee(parent *types.Header) *big.Int {
```

Comment:
Now this function is only used for verification and chain generation(for testing). I think you also need to initialize the baseFee in miner package. https://github.com/ethereum/go-ethereum/blob/e43ac53264429e430450fc1054ae294900172e0d/miner/worker.go#L880

Reply:
Yes agreed - that was done [here](https://github.com/quilt/go-ethereum/blob/d0ce0974028c104da8f3a93c9d6a3dea7eee8b36/miner/worker.go#L912-L917). Is that something you'd also like to see in the PR? I've shied away from including anything here beyond the bare minimum to validate a EIP-1559 activated chain, but happy to include more.

Reply:
IMO I think I would prefer to include it. So if the Aleut block is not defined these rules should never be activated(I think it's safe to add). Otherwise we just have a incomplete functinality.

Reply:
Yes, I just checked your miner's branch. Now I get it. Because of the new mechanism, the ordering and some miner internal needs to be changed. Now I'm fine with it. We can have two PRs for the entire 1559.